File: java/jimmer-cloud/model/src/main/java/org/babyfish/jimmer/example/cloud/model/BookStore.java
package org.babyfish.jimmer.example.cloud.model;

import org.babyfish.jimmer.example.cloud.model.common.BaseEntity;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@Entity(microServiceName = "store-service")
public interface BookStore extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @Key
    String name();

    @Nullable
    String website();

    @OneToMany(mappedBy = "store")
    List<Book> books();
}

File: kotlin/save-command-kt/src/main/kotlin/org/babyfish/jimmer/example/save/model/BookStore.kt
package org.babyfish.jimmer.example.save.model

import org.babyfish.jimmer.sql.*

@Entity
@KeyUniqueConstraint
interface BookStore {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Key
    val name: String

    val website: String?

    @OneToMany(mappedBy = "store")
    val books: List<Book>
}